smart_open/smart_open_lib.py

```diff
@@ -902,7 +903,7 @@ def s3_iter_bucket_process_key(key, retries=3):
             pass
 
 
-def s3_iter_bucket(bucket, prefix='', accept_key=lambda key: True, key_limit=None, workers=16, retries=3):
+def s3_iter_bucket(bucket, prefix='', accept_key=lambda key: True, key_limit=None, workers=16, retries=3, ordered=False, maxresults=None):
```

Comment:
please add this arguments to docstrings with description, what is it

Reply:
I think that maxresults is a poor solution to the problem of the producer queues filling up, because it:

1. Requires the user to know about the implementation details
2. Requires the user to specify this parameter in order to avoid running out of memory while iterating (something that should never really happen, because that is a key benefit of iteration).
3. Terminates the iteration early, without yielding all possible results

I think a better solution would be to process the keys in batches. 

- Make a constant `BATCH_SIZE = 1000` and read up to BATCH_SIZE keys from the iterable at a time.
- Process the batch using imap
- Yield the keys one by one
- Move on with the next batch

I think this approach addresses the three problems I described above.

Reply:
Also, please explain how maxresults is different from key_limit. They seem very strongly related (if not the same) and it seems redundant keeping them both.